### ISSUE TITLE
Incorrect documentation for `add-to-path`

### ISSUE THREAD

**Issue by Lastique** [cygwin/cygwin-install-action#19]
The readme says

> By default, this action prepends Cygwin's /usr/bin directory to the PATH.

This is not actually true since the action *appends* the bin path to PATH, as can be seen [here](https://github.com/cygwin/cygwin-install-action/blob/3bb2cf578250d22bed46555cbf300b22266ef287/action.yml#L102). As a consequence, executables that are found in earlier paths in PATH are preferred over those provided by Cygwin. For example, `g++` will be picked from a MinGW-w64 installation on the `windows-2022` image instead of Cygwin.

**Comment by jon-turney** [comment]
> This is not actually true since the action _appends_ the bin path to PATH, as can be seen [here](https://github.com/cygwin/cygwin-install-action/blob/3bb2cf578250d22bed46555cbf300b22266ef287/action.yml#L102).

Are you sure? 

According to the docs, paths are written to GITHUB_PATH are prepended: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#adding-a-system-path

**Comment by jon-turney** [comment]
> package name was incorrect and the installer didn't fail

Yeah, unfortunately (for historical reasons), we only warn, not fail on unknown package names.  It would be nice to fix that, but probably involves adding yet another option to the installer.

**Comment by Lastique** [comment]
> > This is not actually true since the action _appends_ the bin path to PATH, as can be seen [here](https://github.com/cygwin/cygwin-install-action/blob/3bb2cf578250d22bed46555cbf300b22266ef287/action.yml#L102).
> 
> Are you sure?
> 
> According to the docs, paths are written to GITHUB_PATH are prepended: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#adding-a-system-path

Interesting, I didn't know that. Thanks.

However, the command in the `cygwin-install-action` *appends* to the file identified with `$GITHUB_PATH`, so any paths already listed in that file would be searched first. I think, the documentation could simply say that Cygwin's path is *appended* to `$GITHUB_PATH`, and then maybe link the page you linked for further clarification.

**Comment by Lastique** [comment]
> It would be nice to fix that, but probably involves adding yet another option to the installer.

It would be nice to fix this, indeed. I think, the option should not be a problem since it would be specified in `cygwin-install-action` anyway.

**Comment by jon-turney** [comment]
> It would be nice to fix this, indeed. I think, the option should not be a problem since it would be specified in `cygwin-install-action` anyway.

I created #20 for that issue.

The problem is, as usual, not the technical details, but that someone has to write and test the change.

**Comment by jon-turney** [comment]
> However, the command in the `cygwin-install-action` _appends_ to the file identified with `$GITHUB_PATH`, so any paths already listed in that file would be searched first. I think, the documentation could simply say that Cygwin's path is _appended_ to `$GITHUB_PATH`, and then maybe link the page you linked for further clarification.

I think that the contents of $GITHUB_PATH are prepended to the PATH and then cleared by the runner after every step, so this should be working as expected.

Mentioning GITHUB_PATH in our documentation is probably unhelpful, since this means nothing to persons unfamilar with the details of GitHub runners, might change due to runner changes, and doesn't tell them how we change PATH, which is what they are probably interested in.

I'm going to close this issue, since this discussion seems entirely theoretical at the moment. 

If there's actually a problem, please re-open this with an example of a workflow where you think the PATH should be X:Y:Z after a given step, but it's actually Y:X:Z
